### PR TITLE
wallet, rpc: add wtxid in WalletTxToJSON

### PR DESCRIPTION
**Pull request by brunoerg**
This PR add `wtxid` in `WalletTxToJSON` which allows to return this field in `listsinceblock`, `listtransactions` and `gettransaction` (RPCs).

### CODE REVIEW

**Comment by w0xlt** [comment]
Concept ACK

**Review by luke-jr — CHANGES_REQUESTED** [review]
Needs docs updated

**Comment by brunoerg** [comment]
Done, thanks @luke-jr.

File: src/wallet/rpc/transactions.cpp [hunk 1]
```diff
@@ -33,7 +33,9 @@ static void WalletTxToJSON(const CWallet& wallet, const CWalletTx& wtx, UniValue
         entry.pushKV("trusted", CachedTxIsTrusted(wallet, wtx));
     }
     uint256 hash = wtx.GetHash();
+    uint256 witness_hash = wtx.GetWitnessHash();
     entry.pushKV("txid", hash.GetHex());
+    entry.pushKV("wtxid", witness_hash.GetHex());
```
Comment (achow101) [inline src/wallet/rpc/transactions.cpp:38]: In fe65dab3dbef51e49b484e155ffbf7fd01ea2e8e "wallet: add wtxid in WalletTxToJSON"

nit: `witness_hash` is not used anywhere else, so it does not need to be a separate variable

```suggestion
    entry.pushKV("wtxid", wtx.GetWitnessHash().GetHex());
```

**Review by achow101 — COMMENTED** [review]
ACK 96d2cab3b06639c4ab4fa79b72aa81bdefe03492

File: src/wallet/rpc/transactions.cpp (hunk 1 above)
Comment (brunoerg) [reply src/wallet/rpc/transactions.cpp]: Done!

**Comment by brunoerg** [comment]
force-pushed addressing @achow101's comment.

**Comment by luke-jr** [comment]
Eh? Still don't see docs updated in the diff...

**Comment by brunoerg** [comment]
> Eh? Still don't see docs updated in the diff...

@luke-jr I created `doc/release-notes-24198.md`, isn't it the right way? See https://github.com/bitcoin/bitcoin/blob/master/doc/developer-notes.md#release-notes

**Comment by luke-jr** [comment]
Look in `src/wallet/rpc/transactions.cpp` for:

```c++
    return RPCHelpMan{"gettransaction",
                "\nGet detailed information about in-wallet transaction <txid>\n",
                {
                    {"txid", RPCArg::Type::STR, RPCArg::Optional::NO, "The transaction id"},
```

You need to add "wtxid" here too. Same for the other affected RPC methods.

**Comment by brunoerg** [comment]
> You need to add "wtxid" here too. Same for the other affected RPC methods.

@luke-jr Sorry, my bad. I was with the release notes in my mind, I forgot that one. Force pushed including it. Thank you!

**Review by luke-jr — APPROVED** [review]
utACK

**Comment by brunoerg** [comment]
Force-pushed - rebasing on master now CI works.

**Comment by achow101** [comment]
ACK 79d6764c133638c7feccf4fb1f98242a9ba0863e

**Comment by MarcoFalke** [comment]
Is there a specific use case or is this just for fun?

**Comment by brunoerg** [comment]
> Is there a specific use case or is this just for fun?

Not sure if there is a specific use case. I use `listsinceblock` a lot in a project (and I know many exchanges use it a lot as well, maybe for deposits) and noticed that it didn't return the wtxid which could help me when analyzing transactions.

File: doc/release-notes-24198.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+Updated RPCs
+------------
+
+- The following RPCs: `listtransactions`, `gettransaction` and `listsinceblock`
+  now include the `wtxid` of the transaction.
```
Comment (luke-jr) [inline doc/release-notes-24198.md:5]: nit: Probably a better way to phrase this

File: src/wallet/rpc/transactions.cpp [hunk 3]
```diff
@@ -431,6 +432,7 @@ static const std::vector<RPCResult> TransactionDescriptionString()
            {RPCResult::Type::NUM, "blockindex", /*optional=*/true, "The index of the transaction in the block that includes it."},
            {RPCResult::Type::NUM_TIME, "blocktime", /*optional=*/true, "The block time expressed in " + UNIX_EPOCH_TIME + "."},
            {RPCResult::Type::STR_HEX, "txid", "The transaction id."},
+           {RPCResult::Type::STR_HEX, "wtxid", "The witness transaction id."},
```
Comment (luke-jr) [inline src/wallet/rpc/transactions.cpp:435]: nit: The two existing "wtxid" descriptions are different.

I prefer `"hash of serialized transaction, including witness data"`

**Review by luke-jr — APPROVED** [review]
re-utACK 79d6764c133638c7feccf4fb1f98242a9ba0863e

nit: IMO squash 79d6764c133 (RPC docs) and a5b66738f19 (QA) into e8c659a2970 (feature addition)

File: src/wallet/rpc/transactions.cpp (hunk 3 above)
Comment (brunoerg) [reply src/wallet/rpc/transactions.cpp:435]: I prefer `"hash of serialized transaction, including witness data"` as well, going to change it.

File: doc/release-notes-24198.md (hunk 2 above)
Comment (brunoerg) [reply doc/release-notes-24198.md:5]: Hm, interesting. Maybe:

"Every transaction returned by the following RPCs: `listtransactions`, `gettransaction` and `listsinceblock` now include a `wtxid` field (hash of serialized transaction, including witness data)."

File: src/wallet/rpc/transactions.cpp (hunk 3 above)
Comment (taveechaimekwan) [reply src/wallet/rpc/transactions.cpp:435]: # []()

File: doc/release-notes-24198.md (hunk 2 above)
Comment (luke-jr) [reply doc/release-notes-24198.md:5]: Suggest "The `listtransactions`, `gettransaction`, and `listsinceblock` RPC methods now include a `wtxid` field (hash of serialized transaction, including witness data) for each transaction."

**Comment by brunoerg** [comment]
Force-pushed addressing @luke-jr's comments

**Review by w0xlt — APPROVED** [review]
crACK 7abd8b2

**Comment by achow101** [comment]
re-ACK 7abd8b21ba34f6a42db2cd2d59a4c0e08561f609

**Comment by achow101** [comment]
@luke-jr re-review this?

**Review by luke-jr — APPROVED** [review]
re-utACK 7abd8b21ba34f6a42db2cd2d59a4c0e08561f609

**Pull request merged**